.github/workflows/main.yml

```diff
@@ -0,0 +1,30 @@
+# .github/workflows/main.yml
+
+name: CI
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  test-ruby:
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      matrix:
+        os: [ubuntu]
+        ruby-version: ["2.6", "2.7", "3.0", "3.1", "3.2"]
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
You may want to bump this to `v4` - https://github.com/actions/checkout/releases/tag/v4.0.0

Will avoid node version deprecation warning on CI runs.